apps/tangle-dapp/app/bridge/hooks/useFees.ts

```diff
@@ -0,0 +1,328 @@
+          break;
+      }
+
+      if (fee === null) {
+        return null;
+      }
+
+      return {
+        amount: fee,
+        isLoading,
+        symbol: selectedSourceChain.nativeCurrency.symbol,
+      };
+    }
+
+    function calculateGasFee(): BridgeFeeItem | null {
+      let fee = null;
+      let isLoading = false;
+
+      switch (bridgeType) {
+        case BridgeType.HYPERLANE_EVM_TO_EVM:
+          fee = hyperlaneFees?.gasFee.amount ?? null;
+          isLoading = isLoadingHyperlaneFees;
+          break;
+        case BridgeType.SYGMA_EVM_TO_EVM:
+        case BridgeType.SYGMA_EVM_TO_SUBSTRATE:
+          if (ethersGasPrice && evmEstimatedGasPrice) {
+            fee = ethersGasPrice.times(evmEstimatedGasPrice);
+          }
+          isLoading = isLoadingEthersGasPrice || isLoadingEvmEstimatedGasPrice;
+          break;
+        case BridgeType.SYGMA_SUBSTRATE_TO_SUBSTRATE:
+        case BridgeType.SYGMA_SUBSTRATE_TO_EVM:
+          fee = substrateEstimatedGasFee ?? null;
+          isLoading = isLoadingSubstrateEstimatedGasFee;
+          break;
+      }
+
+      if (fee === null) {
+        return null;
+      }
+
+      return {
+        amount: fee,
+        isLoading,
+        symbol: selectedSourceChain.nativeCurrency.symbol,
+      };
+    }
+
+    function calculateInterchainFee(): BridgeFeeItem | null {
+      if (bridgeType !== BridgeType.HYPERLANE_EVM_TO_EVM || !hyperlaneFees) {
+        return null;
+      }
+
+      return {
+        amount: hyperlaneFees.interchainFee.amount,
+        isLoading: isLoadingHyperlaneFees,
+        symbol: hyperlaneFees.interchainFee.symbol,
+      };
+    }
+
+    // Bridge Fee
+    const bridgeFee = calculateBridgeFee();
+    updateFeeItem('bridge', bridgeFee);
+
+    // Gas Fee
+    const gasFee = calculateGasFee();
+    updateFeeItem('gas', gasFee);
+
+    // Interchain Fee (only for Hyperlane)
+    const interchainFee = calculateInterchainFee();
+    updateFeeItem('interchain', interchainFee);
+  }, [
+    bridgeType,
+    evmSygmaFee,
+    substrateSygmaFee,
+    ethersGasPrice,
+    evmEstimatedGasPrice,
+    substrateEstimatedGasFee,
+    hyperlaneFees,
+    updateFeeItem,
+    selectedSourceChain.nativeCurrency.symbol,
+    isLoadingEvmSygmaFee,
+    isLoadingSubstrateSygmaFee,
+    isLoadingHyperlaneFees,
+    isLoadingEthersGasPrice,
+    isLoadingEvmEstimatedGasPrice,
+    isLoadingSubstrateEstimatedGasFee,
+  ]);
+
+  const isLoading = useMemo(() => {
+    return (
+      isLoadingEvmSygmaFee ||
+      isLoadingSubstrateSygmaFee ||
+      isLoadingEthersGasPrice ||
+      isLoadingEvmEstimatedGasPrice ||
+      isLoadingSubstrateEstimatedGasFee ||
+      isLoadingHyperlaneFees
+    );
+  }, [
+    isLoadingEvmSygmaFee,
+    isLoadingSubstrateSygmaFee,
+    isLoadingEthersGasPrice,
+    isLoadingEvmEstimatedGasPrice,
+    isLoadingSubstrateEstimatedGasFee,
+    isLoadingHyperlaneFees,
+  ]);
+
+  return {
+    isLoading,
+  };
```

Comment:
Would it be more appropriate to directly return `isLoading` instead of an object?